pkg/cluster/k8sres.go

```diff
@@ -960,7 +964,9 @@ func (c *Cluster) getPodEnvironmentConfigMapVariables() ([]v1.EnvVar, error) {
 		}
 	}
 	for k, v := range cm.Data {
-		configMapPodEnvVarsList = append(configMapPodEnvVarsList, v1.EnvVar{Name: k, Value: v})
+		if !isEnvVarPresent(configMapPodEnvVarsList, k) {
+			configMapPodEnvVarsList = append(configMapPodEnvVarsList, v1.EnvVar{Name: k, Value: v})
+		}
```

Comment:
With the result from `getPodEnvironmentConfigMapVariables` we can do the `isEnvVarPresent` in  `generateStatefulSet`, too. This way, the functions will just do what its name tells us it's supposed to do.

Reply:
Fix logic, please review.